llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -29,6 +29,39 @@
 
 using namespace llvm;
 
+static bool CC_AMDGPU_Custom_I1(unsigned ValNo, MVT ValVT, MVT LocVT,
+                                CCValAssign::LocInfo LocInfo,
+                                ISD::ArgFlagsTy ArgFlags, CCState &State) {
+  static bool IsWave64 = static_cast<const GCNSubtarget &>(
+                             State.getMachineFunction().getSubtarget())
+                             .isWave64();
+
+  static const MCPhysReg I1RegList1[] = {
+      AMDGPU::SGPR0_SGPR1,   AMDGPU::SGPR2_SGPR3,   AMDGPU::SGPR4_SGPR5,
```

Comment:
Do we want to only pass in even aligned registers? We could also include all the aliasing odd-based cases

Reply:
By "odd-based cases" do you mean something like SGPR1_SGPR2? Do you have an example where such reg allocation is used?

Reply:
Yes. I mean if you had (i32 inreg %arg0, i64 inreg %arg1), you could use arg0=s0, arg1=s[1:2]

Reply:
This issue deals with i1 arg/return only. Inreg args are handled separately. If you are thinking about a mixed case as follows:
```
foo (i32 inreg %arg0, i1 %arg1)
```
Currently for gfx900,  %arg0 is assigned to s4, and %arg1 to s[6:7]. If in this case you want %arg1 to be given s[5:6], I suppose the list of registers can be changed from `{ ... sgpr4_sgpr5, sgpr6_sgpr7, ...}` to `{... sgpr4_sgpr5, sgpr5_sgrp6, sgpr6_sgrp7, ...}`.
Is this what you had in mind?

Reply:
@arsenm Pls let me know if the above is what you wanted.

Reply:
Yes, the i64 is physically what is being passed, but doesn't have the alignment requirement in the argument list.

Reply:
It seems pairs of SGPRs that start with an odd number, e.g., SGPR5_SGPR6, are
not defined. Only pairs like SGPR4_SGPR5 are defined.

Reply:
Yes, you have to unroll them into the 32-bit pieces and reassemble the virtual register